src/AI/Muriel.java

```diff
@@ -0,0 +1,750 @@
+  }
+
+  @Override
+  public void endTurn()
+  {
+    log(String.format("[======== Muriel ending turn %s for %s =========]", turnNum, myCo));
+    logger = new StringBuffer();
+  }
+
+  private void log(String message)
+  {
+    System.out.println(message);
+    logger.append(message).append('\n');
+  }
+
+  @Override
+  public GameAction getNextAction(GameMap gameMap)
+  {
+    // If we have actions ready to go, don't bother calculating anything.
+    if( !queuedActions.isEmpty() )
+    {
+      GameAction action = queuedActions.poll();
+      log(String.format("  Action: %s", action));
+      return action;
+    }
+
+    // Handle Unit Actions
+    for( Unit unit : myCo.units )
+    {
+      if( unit.isTurnOver ) continue; // Ignore stale units.
+
+      // If we are capturing something, finish what we started.
+      if( unit.getCaptureProgress() > 0 )
+      {
+        log(String.format("%s is currently capturing; continue", unit.toStringWithLocation()));
+        queuedActions.offer(new GameAction.CaptureAction(gameMap, unit, Utils.findShortestPath(unit, unit.x, unit.y, gameMap)));
+        break;
+      }
+
+      //////////////////////////////////////////////////////////////////
+      // Figure out if we should go resupply.
+      boolean shouldResupply = false;
+      // If we are low on fuel.
+      if( unit.fuel < (unit.model.maxFuel/4.0) )
+      {
+        log(String.format("%s is low on fuel.", unit.toStringWithLocation()));
+        shouldResupply = true;
+      }
+      // If we are low on HP, go heal.
+      if( unit.getHP() < 6 ) // Arbitrary threshold
+      {
+        log(String.format("%s is damaged (%s HP).", unit.toStringWithLocation(), unit.getHP()));
+        shouldResupply = true;
+      }
+      // If we are out of ammo.
+      if( unit.weapons != null && unit.weapons.length > 0 )
+      {
+        for( Weapon weap : unit.weapons )
+        {
+          if(weap.ammo == 0)
+          {
+            log(String.format("%s is out of ammo.", unit.toStringWithLocation()));
+            shouldResupply = true;
+          }
+        }
+      }
+      if( shouldResupply )
+      {
+        ArrayList<XYCoord> stations = AIUtils.findRepairDepots(unit);
+        Utils.sortLocationsByDistance(new XYCoord(unit.x, unit.y), stations);
+        for( XYCoord coord : stations )
+        {
+          Location station = gameMap.getLocation(coord);
+          // Go to the nearest unoccupied friendly space, but don't gum up the production lines.
+          if( station.getResident() == null && (station.getEnvironment().terrainType != TerrainType.FACTORY) )
+          {
+            // Plot a course towards a repair station, but only apply the action if it moves us.
+            // If a unit is stuck on the front lines and can't get away past reinforcements, just gotta knuckle up.
+            GameAction goHome = AIUtils.moveTowardLocation(unit, coord, gameMap);
+            if( !goHome.getMoveLocation().equals(new XYCoord(unit.x, unit.y)) )
```

Comment:
This check causes badly-wounded units to leave a repair station they're on and seek a new one.
It's arguably a good thing... but definitely arguable.

Reply:
Yup, not intentional. Good catch.